### PR TITLE
Made possible to use custom templates

### PR DESCRIPTION
**Pull request by velo**
---

### Description

Added the configuration for manually selecting templates.  This would allow anyone that needs/wants to modify a output, to do it by themselves.

For instance, I'm would like to use java record with lombok annotations, by using a custom template I can do it.

Related to #146 and potentially many others

---

Changes were made to:
- [x] Codegen library - Java
- [x] Codegen library - Kotlin
- [x] Codegen library - Scala
- [x] Maven plugin
- [x] Gradle plugin
- [x] SBT plugin

If the change seems reasonable, I will make the gradle plugin and SBT changes

### CODE REVIEW

File: src/main/java/com/kobylynskyi/graphql/codegen/generators/FreeMarkerTemplateFilesCreator.java [hunk 1]
```diff
@@ -48,7 +48,13 @@ public static File create(MappingContext mappingContext,
         }
 
         try (FileWriter fileWriter = new FileWriter(javaSourceFile)) {
-            Template template = FreeMarkerTemplatesRegistry.getTemplateWithLang(language, templateType);
+            Template template;
+            if (mappingContext.getCustomTemplates().containsKey(templateType)) {
+              template = FreeMarkerTemplatesRegistry.getCustomTemplates(mappingContext.getCustomTemplates().get(templateType));
```
Comment (kobylynskyi) [inline src/main/java/com/kobylynskyi/graphql/codegen/generators/FreeMarkerTemplateFilesCreator.java:53]: 1 - Don't make 2 map lookups (first `.containsKey()`, then `.get()`). Instead - do `.get()` and then check if the value is `null`.
2 - Use constant indentation of 4 spaces.

File: src/main/java/com/kobylynskyi/graphql/codegen/model/MappingConfig.java [hunk 2]
```diff
@@ -1,5 +1,7 @@
 package com.kobylynskyi.graphql.codegen.model;
 
+import com.kobylynskyi.graphql.codegen.generators.FreeMarkerTemplateType;
+import java.util.EnumMap;
```
Comment (kobylynskyi) [inline src/main/java/com/kobylynskyi/graphql/codegen/model/MappingConfig.java:4]: Unused import

File: src/main/java/com/kobylynskyi/graphql/codegen/model/MappingConfig.java [hunk 3]
```diff
@@ -241,6 +245,28 @@ public void setCustomTypesMapping(Map<String, String> customTypesMapping) {
...
+     * Put custom template if absent.
+     *
+     * @param from the from
+     * @param to   the to
+     */
+    public void putCustomTemplatesIfAbsent(FreeMarkerTemplateType from, String to) {
+      if (customTemplates == null) {
+        customTemplates = new HashMap<>();
+      }
+      customTemplates.computeIfAbsent(from, k -> to);
```
Comment (kobylynskyi) [inline src/main/java/com/kobylynskyi/graphql/codegen/model/MappingConfig.java:258]: Use 4 spaces indentation across all files

File: src/test/java/com/kobylynskyi/graphql/codegen/GraphQLCodegenAnnotationsTest.java [hunk 4]
```diff
@@ -321,4 +322,14 @@ private void generate(String path) throws IOException {
                 TestUtils.getStaticGeneratedInfo(mappingConfig)).generate();
     }
 
+    @Test
+    void generate_CustomTemplates_Type() throws Exception {
```
Comment (kobylynskyi) [inline src/test/java/com/kobylynskyi/graphql/codegen/GraphQLCodegenAnnotationsTest.java:326]: This test case is not related to annotations, so please move it to a separate test class `GraphQLCodegenCustomTemplatesTest`

**Comment by kobylynskyi** [comment]
Hey @velo 
Thanks for the contribution!
I like this feature.
Please address the comments I put in individual files. 
Also here are additional notes:
1 - Please fix the build, for some reason maven codegen fails with NPE: https://github.com/kobylynskyi/graphql-java-codegen/actions/runs/4445081094/jobs/7821033595?pr=1048
2 - If you can append a new record to the end of the configs description table in `codegen-options.md`.
3 - If you are familiar with Gradle/SBT plugins, please add support to those too.

Thank you!

File: src/main/java/com/kobylynskyi/graphql/codegen/model/MappingConfig.java [hunk 5]
```diff
@@ -241,6 +245,28 @@ public void setCustomTypesMapping(Map<String, String> customTypesMapping) {
         this.customTypesMapping = customTypesMapping;
     }
 
+    /**
+     * Put custom template if absent.
```
Comment (kobylynskyi) [inline src/main/java/com/kobylynskyi/graphql/codegen/model/MappingConfig.java:249]:
```suggestion
     * Provide a path to a custom template for the specific FreeMarker template type (if absent).
```

**Review by kobylynskyi — CHANGES_REQUESTED** [review]

**Review by kobylynskyi — DISMISSED** [review]

**Comment by velo** [comment]
Fixed scalafmt

**Review by kobylynskyi — APPROVED** [review]

**Comment by velo** [comment]
Found a small buh on this, hope I can patch for the release

**Pull request merged**